### ISSUE TITLE
Accessibility issues create "High" severity issues in WCAG 2.1 testing

### ISSUE THREAD

**Issue by tommed** [nathanreyes/v-calendar#1297]
An independent WCAG 2.1 test was carried out on our app, and the following issue was raised in regard to us using v-calendar:

**Background:** All content should be intuitively interact-able using a keyboard interface alone,
with or without a screen reader active. This ensures that users with mobility impairments
who rely on keyboard navigation as they cannot use a mouse can still use the application,
including those who rely on a screen reader.

**Current behaviour:** The date picker widget within the dashboard presents a few issues for
keyboard-only users, particularly when using a screen reader. The following issues are
present:
 * No on-screen instruction regarding shortcut keys for interaction, e.g. home/end for
week navigation, page up/down for month navigation, alt+page up/down for year
navigation.
 * The above-mentioned shortcut keys do not work whilst using a screen reader.
 * Keyboard focus cannot be moved to the previous and next navigation buttons, nor the
month selection button between the previous/next buttons
 * The extended month/year picking functionality achieved by clicking the currently
displayed month text is not accessible via keyboard – all relevant state information and
control markup is not exposed to assistive technologies.

**To replicate:** Using the tab key, attempt to move focus to the previous/next buttons within
the date picker widget. Using a screen reader, attempt to navigate the date picker widget and try to switch months.

**Expected behaviour:** The date picker widget is fully keyboard accessible – all on-screen
buttons receive focus and interact with keyboard. On-screen instructions for shortcut keys are provided, which are compatible with screen readers. Users can also opt to manually enter the date rather than relying on the date picker.

**Recommended fix:** Provide an accessible alternative to the date picker input by allowing the
user to manually enter the date / date range of their filter. The input fields should be
labelled visually and programmatically as standard, and any validation should be
implemented accessibly (aria-live for inline validation, programmatic association of errors to
form fields).

In addition, and at the very least, ensure that the previous and next buttons are focusable
via the tab key as per standard keyboard navigation, by using correct button mark-up, and
ensure that they are labelled both programmatically, and visually using tooltips, which
display on hover and focus of the previous/next buttons

**Comment by tommed** [comment]
Is this component unsupported then?

**Comment by tommed** [comment]
From the testers:

> ...the issues described impacted screen reader users as that’s the case the issue would need to remain as a _high severity_. The keyboard interaction will be impacted when the screen reader is running vs without as the SR will capture the keystrokes.

**Comment by drogers98** [comment]
Not 100% the same issue, just wanted to point out that the PR here does fix some more "general" accessibility issues with the prev/next buttons: https://github.com/nathanreyes/v-calendar/pull/1089

Would be a good start, if you are seeing this issue because of accessibility check failure on v-calendar.

**Comment by tommed** [comment]
Thanks. Tbh, I think there's a lot more than needs doing than previous/next buttons for accessibility. I really hope it gets sorted at some point because otherwise this is a great component. We just can't continue to use for professional projects as there are legal requirements in some countries to provide levels of accessibility which this falls short of.
